.github/workflows/python-wheels.yaml

```
@@ -0,0 +1,97 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Build Python Wheels
+
+# Build wheels weekly or when requested
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/python-wheels.yaml'
+  workflow_dispatch:
+  schedule:
+    - cron: '6 0 * * 0'
+
+jobs:
+  build_sdist:
+    runs-on: "ubuntu-20.04"
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v3
+    - name: Check that cmake is installed
+      run: |
+        cmake --version
+
+    - name: Install build
+      run: |
+        pip install build
+
+    - name: Build sdist
+      run: |
+        cd python
+        python -m build --sdist
+
+    - name: Check install from sdist
+      run: |
+        pip install python/dist/nanoarrow-*.tar.gz
+
+    - name: Test import
+      run: |
+        python -c "import nanoarrow; print(nanoarrow.__version__)"
+
+    - uses: actions/upload-artifact@v3
+      with:
+        name: sdist
+        path: ./python/dist/nanoarrow-*.tar.gz
+
+  build_wheels:
+    needs: ["build_sdist"]
+    name: Build wheels on ${{ matrix.config.label }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+          - {os: "ubuntu-20.04", label: "linux"}
+          - {os: "windows-2019", label: "windows"}
+          - {os: "macOS-11", label: "macOS"}
+          # Uncomment when apache/arrow-nanoarrow is added to the arrow runner group
+          # - {os: ["self-hosted", "arm"], label: "linux/arm64"}
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v3
+
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel==2.15.0
+
+      - name: Build wheels
+        run: |
+          python -m cibuildwheel --output-dir wheelhouse python
```

```suggestion
      - name: Build wheels
        uses: pypa/cibuildwheel@v2.16.2
```

I am not sure why the cibuildwheel README shows the manual way, but the docs recommend using the action (https://cibuildwheel.readthedocs.io/en/stable/setup/#github-actions)